src/token_adapter/jwt.ts

```
@@ -1,57 +1,125 @@
 import {ITokenData, TokenType} from '@essential-projects/core_contracts';
 import {ITokenAdapter} from '@essential-projects/iam_contracts';
-import {sign, SignOptions, verify} from 'jsonwebtoken';
+import {IPkiService} from '@essential-projects/pki_service_contracts';
 
-import * as BluebirdPromise from 'bluebird';
-
-const secret = 'secret';
+import {sign, SignOptions, verify, VerifyOptions} from 'jsonwebtoken';
 
 export class JwtTokenAdapter implements ITokenAdapter {
 
+  private _pkiService: IPkiService = undefined;
+  public config: any = undefined;
+
+  constructor(pkiService: IPkiService) {
+    this._pkiService = pkiService;
+  }
+
   public get tokenType(): TokenType {
     return TokenType.jwt;
   }
 
+  private get pkiService(): IPkiService {
+    return this._pkiService;
+  }
+
   public async encode(payload: ITokenData): Promise<string> {
 
-    const options: SignOptions = {};
-    const jwt = await this._sign(payload, secret, options);
-    return <string> jwt;
+    const algorithm: string = this.config && this.config.algorithm ? this.config.algorithm : 'HS256';
+    const expiresIn: string = this.config && this.config.expiresIn ? this.config.expiresIn : undefined;
```

I don't think you need to check for `this.config` every time. checking if the members `algorithm` an `expiresIn` are set is enough.

If you don't set a config, addict-ioc will tell you to set one.
if you set it to something other than an object on purpose, the program should throw an error at you

---

Das Modul sollte sich per default genauso verhalten wie vorher. Und da gibt es Installationen ohne config, die sonst angepasst werden müssten.

---

@ppoetschke wie im kommentar unten:
Dann ist es halt ein breaking change, da Paket bekomme ne neue Version die das reflektiert und gut ist